andela/dark-knight-rc
# #158028143 Implement Wallet System

#### What does this PR do?
This PR adds wallet system with support for:
* Paying for product with wallet,
* Funding wallet wit Paystack, 
* Withdrawing from wallet.

#### Description of Task to be completed?
- Add wallet for every signed-in user
- Remove paystack from checkout 
- Add wallet payment to checkout section
- Implement support for topping up wallet from checkout page
- Add wallet shortcut to account dropdown
- Design wallet page UI
- Design checkout section UI
- Create pop-up confirmation modal 

#### How should this be manually tested?
Wallets are activated by default. However to use the wallet, Paystack needs to be activated.
- As an admin user, 
    * open the dashboard, 
    * click on the credit card icon,
    * toggle the Paystack section
    * Select appropriate options and add keys Paystack provided you
    * Add product to cart, 
    * Enter payement details,
    * Choose a shipping option,
    * Browse to checkout section,
    * Click on the button under `Pay with Wallet`.
- As a regular user, 
    * Paying with Wallet
        * Add product to cart, 
        * Enter payement details,
        * Choose a shipping option,
        * Checkout using wallet.
    * Funding a Wallet 
        * On the navbar, click the section with your name,
        * From the resulting dropdown, click the wallet icon to go to wallet page, 
        * Browse to Fund Wallet section,
        * Enter the amount you want fund the wallet with,
        * Click on the `Fund wallet` button.
    * Withdrawing from wallet
        * On the navbar, click the section with your name,
        * From the resulting dropdown, click the wallet icon to go to wallet page, 
        * Browse to `Withdraw from Wallet` section,
        * Enter the amount you want to withdraw,
        * Click on the `Withdraw` button,
        * An email should be sent to you and the admin.
        

#### What are the relevant pivotal tracker stories?
[158028143](https://www.pivotaltracker.com/story/show/158028143)

#### Screenshots
##### Wallet Page
<img width="1680" alt="wallet page" src="https://user-images.githubusercontent.com/20358651/41671892-5e7d8472-74b0-11e8-91ed-f87647066af7.png">

##### Wallet Funding with Paystack
<img width="1675" alt="wallet funding with paystack" src="https://user-images.githubusercontent.com/20358651/41671891-5e513188-74b0-11e8-8343-2a1675a66d39.png">

##### Wallet Withdraw Notification
<img width="1680" alt="wallet withdraw notification" src="https://user-images.githubusercontent.com/20358651/41671893-5eb3041c-74b0-11e8-8bed-cffe1d8fa3cf.png">

##### Checkout Fund in Wallet
<img width="1680" alt="checkout fund in wallet" src="https://user-images.githubusercontent.com/20358651/41671887-5dbc5950-74b0-11e8-8439-d6d2eb800914.png">

##### Checkout No Fund in Wallet
<img width="1680" alt="checkout no fund in wallet png" src="https://user-images.githubusercontent.com/20358651/41671889-5e054034-74b0-11e8-8862-0e4a1fefdc33.png">
